pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -200,11 +203,12 @@ private String createSegment(String dataFilePath, Schema schema, String seqId) t
       return segmentName;
     }
 
-    private RecordReaderConfig getReaderConfig(FileFormat fileFormat) {
+    private RecordReaderConfig getReaderConfig(FileFormat fileFormat) throws IOException {
       RecordReaderConfig readerConfig = null;
       switch (fileFormat) {
         case CSV:
-          readerConfig = new CSVRecordReaderConfig();
+          readerConfig = (null == _readerConfig) ? new CSVRecordReaderConfig()
+                         : new ObjectMapper().readValue(_readerConfig, CSVRecordReaderConfig.class);
```

Comment:
Can you change this so that if it throws an exception when reading the readerConfig, you log a message stating that CSV reader config __ could not be read?